sql/analyzer/indexed_joins.go

```diff
@@ -536,14 +597,9 @@ func (j joinIndexes) getUsableIndex(schema sql.Schema) *joinIndex {
 }
 
 // schemaContainsField returns whether the schema given has a GetField expression with the column and table name given.
-func schemaContainsField(schema sql.Schema, field *expression.GetField) bool {
-	for _, col := range schema {
-		if strings.ToLower(col.Source) == strings.ToLower(field.Table()) &&
-			strings.ToLower(col.Name) == strings.ToLower(field.Name()) {
-			return true
-		}
-	}
-	return false
+func schemaContainsField(schemaCols map[tableCol]struct{}, field *expression.GetField) bool {
+	_, ok := schemaCols[tableCol{field.Table(), field.Name()}]
```

Comment:
You are omitting the strings.ToLower here, are you sure it's applied higher in the stack?